New notebook and code file for continuum subtraction.

pjjanas opened the pull request:
Also added some descriptions into the notebook.

pjjanas committed:
Added coordinate axes to image plot.

Also added some descriptions into the notebook.

pjjanas committed:
Initial program for continuum subtraction

pjjanas committed:
Misc changes

pjjanas committed:
Notebook for continuum subtraction

pjjanas committed:
Added savefig() option to figure

pjjanas committed:
Fix narrowband image allocation

pjjanas committed:
Update continuum_sub.py

Changing narrowband image from F187N to F444W-F470N as seen in make_RGB jupyter notebook.

pjjanas committed:
Added further code for continuum subtraction

pjjanas committed:
Added code and plot generation

Added code and descriptions on continuum subtractions. Also has plots of subtracted image and sigma clipping also

pjjanas committed:
Fixed coordinate axis labels

pjjanas committed:
Merge branch 'chng' of https://github.com/pjjanas/dias-jwst into chng

pjjanas committed:
Added class ContinuumSubtract()

New class which performs all tasks necessary for continuum subtraction (work in progress)

pjjanas committed:
Added ContinuumSubtract() class implementation and enhanced subtracted image

PaddyKavanagh commented:
Hi Pawel.

Main thing that jumps out at me is with the scale factor:

```python
# Scale continuum data and subtract, then can rerun previous cell for plot
scale_factor = 2
subc = c.continuum_sub(reprojected, scale_factor)
```

Why 2 here? You should initially get the scale_factor (and error) by fitting a straight line to the c_dat_flat, n_dat_flat datapoints, excluding those with c_dat_flat = 0. 

Also, it looks like you are using the full arrays to produce c_dat_flat, n_dat_flat  in the notebook. Try selecting only part of the arrays (one on cloud, one off). It would be nice to show an image with the image region used to produce the scale_factor shown.

pjjanas committed:
Added line fitting

New function in ContinuumSubract() class for line fitting to two specific regions defined by user

pjjanas committed:
Added plots with line fits with updated ContinuumSubtract()

pjjanas commented:
> Hi Pawel.
> 
> Main thing that jumps out at me is with the scale factor:
> 
> ```python
> # Scale continuum data and subtract, then can rerun previous cell for plot
> scale_factor = 2
> subc = c.continuum_sub(reprojected, scale_factor)
> ```
> 
> Why 2 here? You should initially get the scale_factor (and error) by fitting a straight line to the c_dat_flat, n_dat_flat datapoints, excluding those with c_dat_flat = 0.
> 
> Also, it looks like you are using the full arrays to produce c_dat_flat, n_dat_flat in the notebook. Try selecting only part of the arrays (one on cloud, one off). It would be nice to show an image with the image region used to produce the scale_factor shown.
f

Hi Patrick,

I initially set the scale factor to 2 just by eye for the subtraction. I just updated `continuum_sub.py` and the notebook to show the best fit line. I added a function that lets you choose regions desired (one on cloud and one off) and then carry out the line of best fit from that data instead. I'm not really sure (or I forgot) how the slope of the line is used to then get the scale parameter, but the implementation is there for calculating the slope/intercept. I also plotted the regions I used in the notebook.

pjjanas committed:
Added functionability for outlier removal

Added remove_outliers() function to code file and called within ContinuumSubtract.get_line_params(). Uses KMeans clustering algorithm for outlier removal.

pjjanas committed:
Updated  using  and plots

pjjanas committed:
Update README.md

pjjanas committed:
Plotting fil_subc instead of subc

pjjanas committed:
Merge branch 'chng' of https://github.com/pjjanas/dias-jwst into chng
